vm/src/main.rs

```diff
@@ -13,12 +13,11 @@
 
 mod x86_64;
 
+libhypatia::define_segment!();
+
 #[cfg_attr(not(test), start, no_mangle)]
-pub extern "C" fn main() {
```

Comment:
`vm` is a task, not a segment, and ought to have a `main` entry point, not `init`.